pkg/bisect/bisect.go

```diff
@@ -421,13 +421,13 @@ func (env *env) build() (*vcs.Commit, string, error) {
 		return nil, "", fmt.Errorf("kernel clean failed: %v", err)
 	}
 	kern := &env.cfg.Kernel
-	_, kernelSign, err := env.inst.BuildKernel(bisectEnv.Compiler, env.cfg.Ccache, kern.Userspace,
+	_, imageDetails, err := env.inst.BuildKernel(bisectEnv.Compiler, env.cfg.Ccache, kern.Userspace,
```

Comment:
I think here it's better to use imageDetails.CompilerID and remove build.CompilerIdentity. pkg/bisect is also subject to "empty compiler ID" problem.
And when we do this, I would unexport CompilerIdentity function from pkg/build. And new use of CompilerIdentity outside of pkg/build will be subject to the same problem.

Reply:
Made this function local and updated pkg/bisect so that it only relies on the value returned from the builder